src/tests/server.ts

```diff
@@ -1527,6 +1527,56 @@ describe('Subscribe', () => {
 
     await server.waitForComplete();
   });
+
+  it('should use a custom JSON replacer function', async () => {
+    const { url } = await startTServer({
+      schema,
+      replacer: (_, value) => {
+        if (typeof value == 'string') {
+          return value.toUpperCase();
+        }
+        return value;
+      },
+    });
+
+    const client = await createTClient(url);
+    client.ws.send(
+      stringifyMessage<MessageType.ConnectionInit>({
+        type: MessageType.ConnectionInit,
+      }),
+    );
+
+    await client.waitForMessage(({ data }) => {
+      expect(data).toMatchInlineSnapshot(`"{\\"type\\":\\"CONNECTION_ACK\\"}"`);
```

Comment:
Simpler?
```suggestion
      expect(data).toBe('"{"type":"CONNECTION_ACK"}"');
```

Reply:
I used an inline snapshot because I always find getting the quotes and escapes right is hard and I'd rather just `yarn test -u` and let jest write exactly the string that will work. Seems easier to maintain but happy to fix it if you feel strongly!

Reply:
Getting which quotes and escapes right? With the `toBe` there is nothing to escape.

Furthermore, the snapshot here will change only if the test case changes which means you'll have to change the test anyway. I feel you would have a harder time decoding the escapes when looking at the tests, comparing the snapshots, having to do `-u`; in comparison to a plain JSON literal.

If the JSON was large, then it'd be fine; but with these 25 characters, I'd like to keep it inline please.

Reply:
Additionally to getting rid of the escapes, you also drop the extra quotes. So the suggestion is actually:
```suggestion
      expect(data).toBe('{"type":"CONNECTION_ACK"}');
```

Reply:
To each their own, changed!